internal/mempool/v1/reactor.go

```diff
@@ -329,6 +326,8 @@ func (r *Reactor) broadcastTxRoutine(peerID types.NodeID, closer *tmsync.Closer)
 		}
 	}()
 
+	peerMempoolID := r.ids.GetForPeer(peerID)
+	var nextGossipTx *clist.CElement
 	for {
```

Comment:
I don't see the condition where this loop ever breaks in the case that we see another TX

Reply:
It doesn't break in that case—the `nextGossipTx` is just a state variable reused during the lifecycle of the loop. All I did here was move its declaration closer to the use.